src/python/gem5/resources/resource.py

```diff
@@ -554,6 +565,153 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    """
+    A suite resource. This resource is used to specify a suite of workloads to
+    run on a board. It contains a list of workloads to run, along with their
+    IDs and versions.
+
+    Each workload in a suite is used to create a `WorkloadResource` object.
+    These objects are stored in a list and can be iterated over.
+    """
+
+    def __init__(
+        self,
+        workloads: Optional[List[Dict[str, Any]]] = None,
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+        workload_obj_list: Optional[List["WorkloadResource"]] = None,
+        **kwargs,
+    ) -> None:
+        """
+        :param workloads: A list of dictionaries containing the workload
+        :param _workload_obj_list: A list of `WorkloadResource` objects
+        created from the `_workloads` parameter.
+        :param local_path: The path on the host system where this resource is
+        located.
+        :param description: Description describing this resource. Not a
+        required parameter. By default is None.
+        :param source: The source (as in "source code") for this resource
+        on gem5-resources. Not a required parameter. By default is None.
+        :param resource_version: Version of the resource itself.
+        """
+        self._workloads = workloads
+        self._workload_obj_list = workload_obj_list
+        self._local_path = local_path
+        self._description = description
+        self._source = source
+        self._resource_version = resource_version
+
+        super().__init__(
+            local_path=local_path,
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+
+    def __iter__(self) -> Generator["WorkloadResource", None, None]:
+        """
+        Returns a generator that iterates over the workloads in the suite.
+
+        :returns: A generator that iterates over the workloads in the suite.
+        """
+        for workload in self._workload_obj_list:
+            yield workload
+
+    def __len__(self):
+        """
+        Returns the number of workloads in the suite.
+
+        :returns: The number of workloads in the suite.
+        """
+        return len(self._workload_obj_list)
+
+    def with_input_group(self, input_group: str) -> "SuiteResource":
```

Comment:
I think there should be an exception thrown here if an input_group is specified which doesn't exist for this workload. Quite simple logic:

```py
if input_group not in get_input_groups():
    raise Exception("...")
```

I'd have the exception say that the value is invalid and then list the valid values (i.e., `get_input_groups()`).